File: c++/selection_sort.cpp
//#include<bits/stdc++.h>
#include<iostream>

void selectionSort(int arr[], int n){

    for(int i = 0; i < n-1; i++){
        int smallestindx = i; //unsorted part starting
        for(int j = i+1; j < n; j++){
            if(arr[j] < arr[smallestindx]){
                smallestindx = j;
            }
        }

        std::swap(arr[i], arr[smallestindx]);
    }
}
void printArray(int arr[], int n){
    for(int i = 0; i < n; i++){
        std::cout << arr[i] << " ";
    }
}

int main(){
    int n = 5;
    int arr[] = {4, 1, 5, 2, 3};

    selectionSort(arr, n);
    printArray(arr, n);
    return 0;
}

/*  Time complexity: O(n^2) */

File: java/SelectionSort.java
// Implement selection Sort in java
// Time complexity is => O(n^2)
// It select the minimum element from the array an push it the ith index
import java.util.Scanner;
public class SelectionSort {
    public void selectionSorting(int arr[], int size)
    {
        for(int i=0; i<=size-2; i++)
        {
            for(int j=i+1; j!=size; j++)
            {
                if(arr[i]>arr[j])
                {
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp; 
                }
            }
        }   
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the size of the array : ");
        int size = sc.nextInt();
        int arr[] = new int[size];
        System.out.print("Enter the element in the array : ");
        for(int i=0; i<size; i++)
        {
            arr[i] = sc.nextInt();
        }
        SelectionSort s1 = new SelectionSort();
        s1.selectionSorting(arr,size);
        for(int i=0; i<size; i++)
        {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
        sc.close();
    }
}